source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/AdaptiveCardsSharedModelUnitTest.cpp

```diff
@@ -0,0 +1,206 @@
+#include "stdafx.h"
+#include "CppUnitTest.h"
+#include "TextBlock.h"
+#include <time.h>
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+using namespace AdaptiveCards;
+using namespace std;
+
+namespace AdaptiveCardsSharedModelUnitTest
+{        
+    TEST_CLASS(DateAndTimeTest)
+    {
+    public:
+        TEST_METHOD(ISO8601TransformTest)
+        {
+            TextBlock blck;
+            blck.SetText("{{DATE(2017-02-13T20:46:30Z, SHORT)}}");
+            Assert::AreEqual<string>("02/13/17",  blck.GetText());
+        }
+        TEST_METHOD(ISO8601TransformTestLONG)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z, LONG)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("Monday, February 13, 2017", blck.GetText());
+        }
+        TEST_METHOD(ISO8601TransformTestSmallCaseLONG)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z, Long)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("Monday, February 13, 2017", blck.GetText());
+        }
+        TEST_METHOD(ISO8601TransformTestDefault)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("02/13/17", blck.GetText());
+        }
+
+        TEST_METHOD(ISO8601TransformBadInputTest)
+        {
+            TextBlock blck;
+            string testString = "{a{DATE(2017-02-13T20:46:30Z, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{a{DATE(2017-02-13T20:46:30Z, SHORT)}}", blck.GetText());
+        }
+
+        TEST_METHOD(ISO8601TransformBadInputTestMissingClosingCurlyBracket)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z, SHORT)}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{{DATE(2017-02-13T20:46:30Z, SHORT)}", blck.GetText());
+        }
+
+        TEST_METHOD(ISO8601TransformBadInputTestBadYear)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017a02-13T20:46:30Z, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{{DATE(2017a02-13T20:46:30Z, SHORT)}}", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToDateTest1)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("02/13/17", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToDateTest2)
+        {
+            TextBlock blck;
+            string testString = "{{TIME(2017-10-27T18:19:09Z, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("11:19:09 AM", blck.GetText());
```

Comment:
>09 [](start = 44, length = 2)

I believe we shouldn't show the seconds. The api documentation drops the seconds. Could you follow up with Matt to make sure?

Reply:
yes, that's what i thought, but html renderer shows seconds; i will double check

---
In reply to: [149153560](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149153560) [](ancestors = 149153560)